src/ProjectBundle/Utils/SectorService.php

```
@@ -265,4 +266,22 @@ public function getSubsBySector(): iterable
         }
         return $sectors;
     }
+
+
+    /**
+     * @param string $sector
+     *
+     * @return Sector[]
+     * @throws NotFoundException
+     */
+    public function findSubsSectorsBySector(string $sector): array
```

Wrong checking of existstence. In this context, you should throw InvalidArgumentException or its children

---

Also, if object Sector exists in system, you should pass it instead of string.